tf-psa-crypto/tests/suites/test_suite_gcm.function

```diff
@@ -605,6 +605,146 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void gcm_encrypt_input_output_buffer_overlap(int cipher_id, data_t *key_str,
+                         data_t *src_str, data_t *iv_str,
+                         data_t *add_str, data_t *dst,
+                         int tag_len_bits, data_t *tag,
+                         int init_result)
+{
+    unsigned char *buffer;
+    size_t buffer_len;
+    unsigned char tag_output[16];
+    mbedtls_gcm_context ctx;
+    size_t tag_len = tag_len_bits / 8;
+    size_t n1;
+    size_t n1_add;
+
+    BLOCK_CIPHER_PSA_INIT();
+    mbedtls_gcm_init(&ctx);
+
+    /* GCM includes padding and therefore input length can be shorter than the output length
+     * Therefore we must ensure we round up to the nearest 128-bits/16-bytes.
+     */
+    buffer_len = src_str->len;
+    if (buffer_len % 16 != 0) {
+      buffer_len += (16 - (buffer_len % 16));
+    }
+    buffer = malloc(buffer_len);
```

Comment:
Best to use `TEST_CALLOC()` here as it handles all the edge cases:
```suggestion
    buffer = TEST_CALLOC(buffer_len, 1);
```

Reply:
Thanks, fixed now.